### ISSUE TITLE
Add workflow.current_step_uuid() function

### ISSUE THREAD

**Issue by ericl** [ray-project/ray#18356]
Adding `workflow.current_step_uuid()` would simplify writing idempotent workflows. Instead of generating a random ID and checkpointing it in a previous step, the user can simply get the current_step_uuid, which we can guarantee is stably unique across step retries.

**Comment by ericl** [comment]
@iycheng suggested we might automatically make portions of workflows idempotent. For example, suppose you have a workflow step:

```
actor = User.get_or_create("user_1")
actor.notify("hi")
```

We could automatically make inject idempotency tokens to deliver `notify("hi")` exactly once.

**Comment by iycheng** [comment]
@ericl  I think it's going to be fine. We actually have workflow step id, right? What if we just use `workflow.get_current_step_id`?

**Comment by ericl** [comment]
It could work. Just to make sure, is get_current_step_id (1) return the same value across different retries of the step, and (2) globally unique? (might need to prepend the workflow id).

**Comment by iycheng** [comment]
Yes. Globally unique within the step id. We need to prepend the workflow id.

**Comment by iycheng** [comment]
I think we probably need: get_step_token() function and it'll do: workflow id + step_id + actor id + actor method + some deduplication

**Comment by ericl** [comment]
We don't need actor id / method right?

**Comment by iycheng** [comment]
> We don't need actor id / method right?

We need it if we'd like to fix the case of:

```
handler.method.run_async()
handler.method.run_async()
```

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
